bin/certbot-auth.sh.aws

```diff
@@ -0,0 +1,93 @@
+[[ -z "$CERTBOT_DOMAIN" ]] && { echo "error: this script is designed to run through 'certbot ... --manual-auth-hook=$0' (\$CERTBOT_DOMAIN is not set)."; exit -1; }
+
+# CERTBOT_DOMAIN = aas.dirac.institute
+HOST=_acme-challenge.$(echo $CERTBOT_DOMAIN | cut -d . -f 1)
+DOMAIN=$(echo $CERTBOT_DOMAIN | cut -d . -f 2-)
+
+hosted_zone_id=$(aws --output=json route53 list-hosted-zones | jq --arg DOMAIN "${DOMAIN}." ' .HostedZones | .[] | select(.Name==$DOMAIN) | .Id ')
+# remove trailing / preceeding quotes
+hosted_zone_id="${hosted_zone_id%\"}"
+hosted_zone_id="${hosted_zone_id#\"}"
+
+record_dns_name="${HOST}.${DOMAIN}"
+
+resources=$(aws --output=json route53 list-resource-record-sets --hosted-zone-id $hosted_zone_id)
+host_txt_record=$(echo $resources | jq --arg record_dns_name "${record_dns_name}." ' .ResourceRecordSets | .[] | select(.Name==$record_dns_name and .Type=="TXT")')
+
+if [[ $host_txt_record ]]; then
+    name=$(echo $host_txt_record | jq '.Name')
+    type=$(echo $host_txt_record | jq '.Type')
+    ttl=$(echo $host_txt_record | jq '.TTL')
+    resource_records=$(echo $host_txt_record | jq '.ResourceRecords')
+
+    delete_request=$(cat <<EOF
+    {
+        "Comment": "Add certbot validation TXT",   
+        "Changes": [
+            {
+                "Action": "DELETE",
+                "ResourceRecordSet": {
+                    "Name": $name,
+                    "Type": $type,
+                    "TTL": $ttl,
+                    "ResourceRecords": $resource_records
+                }
+            }
+        ]
+    }
+EOF
+)
+    response=$(aws --output=json route53 change-resource-record-sets --hosted-zone-id $hosted_zone_id --change-batch "$delete_request")
+    [[ $response ]] || echo "Error sending request to route 53!"
+fi
+create_request=$(cat <<EOF
+{
+    "Comment": "Add certbot validation TXT",   
+    "Changes": [
+        {
+            "Action": "CREATE",
+            "ResourceRecordSet": {
+                "Name": "${HOST}.${DOMAIN}",
+                "Type": "TXT",
+                "TTL": 1,
+                "ResourceRecords": [{
+                    "Value" : "\"$CERTBOT_VALIDATION\""
+                }]
+            }
+        }
+    ]
+}
+EOF
+)
+response=$(aws --output=json route53 change-resource-record-sets --hosted-zone-id $hosted_zone_id --change-batch "$create_request")
+[[ $response ]] || echo "Error sending request to route 53!"
+
+resources=$(aws --output=json route53 list-resource-record-sets --hosted-zone-id $hosted_zone_id)
+host_txt_record=$(echo $resources | jq --arg record_dns_name "${record_dns_name}." ' .ResourceRecordSets | .[] | select(.Name==$record_dns_name and .Type=="TXT")')
+resource_records=$(echo $host_txt_record | jq '.ResourceRecords')
+value=$(echo $resource_records | jq '.[0] | .Value ')
+
+hdr=
+while [[ $value != "\"\\\"$CERTBOT_VALIDATION\\\"\"" ]]; do
+	resources=$(aws --output=json route53 list-resource-record-sets --hosted-zone-id $hosted_zone_id)
```

Comment:
See if propagation status can be queried via the [https://docs.aws.amazon.com/Route53/latest/APIReference/API_GetChange.html](GetChange) API (`aws route53 get-change`); I think this would eliminate the need for the 60-second wait at the end.